server/spotify_service.go

```diff
@@ -1,31 +1,36 @@
 package server
 
 import (
-	"bytes"
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"log"
 	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/pkg/errors"
-	"github.com/rs/zerolog/log"
+	"github.com/sirupsen/logrus"
 )
 
 const (
 	defaultTimeout = 30
 	scope          = "user-read-recently-played playlist-modify-public playlist-read-collaborative user-read-recently-played user-top-read user-library-read"
-	redirectURL    = "https://bot-bk.website/spotifyCallback"
-
-	AuthState = "spotify-auth-state"
+	redirectURL    = "http://localhost:8080/spotify-callback"
+	AuthState      = "spotify-auth-state"
 )
 
 var (
 	errorUnableToGetToken = errors.New("unable to get token from spotify")
 )
 
 type SpotifyService interface {
-	Login(state string) error
+	GetAuthURL(state string) string
+	GetTokenRequest(uid, code string) error
+	GetTokenResponse(res *http.Response) (string, string, error)
```

Comment:
Maybe there are some other names that's more self-explain than `GetTokenRequest` and `GetTokenResponse`